### PR TITLE
Intern 4 and then some

### PR DESCRIPTION
**Pull request by rorticus**
**Type:** feature

The following has been addressed in the PR:

* [x] There is a related issue
* [x] All code matches the [style guide](https://github.com/dojo/meta/blob/master/STYLE.md)
* [x] Unit or Functional tests are included in the PR

**Description:**

Runs test w/ intern 4! Also..

* Made the link that gets printed out to run the tests clickable (https://github.com/dojo/cli-test-intern/issues/24)
* Cleaned up intern config (https://github.com/dojo/cli-test-intern/issues/38)
* Showing browser URL whether or not the tests failed
* Refactored arguments

Note that this is waiting for a fix from intern to read child configs from node namespaces, as well as https://github.com/dojo/cli-build-webpack/pull/208 and https://github.com/dojo/cli-build-webpack/pull/209 .

Resolves https://github.com/dojo/cli-test-intern/issues/52 , https://github.com/dojo/cli-test-intern/issues/21 , https://github.com/dojo/cli-test-intern/issues/60 , https://github.com/dojo/cli-test-intern/issues/38 , https://github.com/dojo/cli-test-intern/issues/24

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,11 +4,12 @@
...
-/html-report
 /node_modules
 /typings
 .baseDir.ts
 .tscache
 coverage-unmapped.json
 coverage-final.json
 npm-debug.log
+coverage-final.lcov
+coverage/
```
Comment (kitsonk) [inline .gitignore]: Missing newline

Comment (rorticus) [reply .gitignore]: Fixed

File: src/runTests.ts [hunk 2]
```diff
@@ -1,64 +1,97 @@
...
 	}
 
-	args.push(...(reporters ? reporters.split(',').map((reporter) => `reporters=${reporter}`) : [ `reporters=${internReporter}` ]));
+	args.push(...(reporters ? reporters.split(',').map((reporter) => `reporters=${reporter}`) : []));
 
-	if (config === 'testingbot' && testingKey && secret) {
-		args.push(`tunnelOptions={ "verbose": "true", "apiKey": "${testingKey}", "apiSecret": "${secret}" }`);
-		args.push(`webdriver={ "host": "http://hub.testingbot.com/wd/hub", "username": "${testingKey}", "accessKey": "${secret}" }`);
+	if (childConfig === 'testingbot' && testingKey && secret) {
+		args.push(`tunnelOptions={ "verbose": "true", "hostname": "hub.testingbot.com", "apiKey": "${testingKey}", "apiSecret": "${secret}" }`);
```
Comment (jason0x43) [inline src/runTests.ts]: The TestingBot tunnel uses the same auth credentials as the other tunnels (username and accessKey).

File: src/runTests.ts [hunk 3]
```diff
@@ -1,64 +1,97 @@
...
 	}
 	else if (userName && testingKey) {
-		args.push(`tunnelOptions={ "username": "${userName}", "accessKey": "${testingKey}" }`);
+		args.push(`tunnelOptions={ "username": "${userName}", "apiKey": "${testingKey}" }`);
```
Comment (jason0x43) [inline src/runTests.ts]: `accessKey` rather than `apiKey`

File: tests/intern.ts [hunk 4]
```diff
@@ -1,8 +1,3 @@
-export const proxyPort = 9000;
```
Comment (jason0x43) [inline tests/intern.ts]: Is this file still necessary?

File: tests/unit/runTests.ts [hunk 5]
```diff
@@ -48,29 +43,17 @@ describe('runTests', () => {
...
 			verbose: true
 		});
-		assert.strictEqual(consoleStub.callCount, 5);
-		assert.include(consoleStub.getCall(0).args[0], 'testing "');
-		assert.include(consoleStub.getCall(1).args[0], 'Parsed arguments for intern:');
-		assert.include(consoleStub.getCall(2).args[0], `config=${path.join('intern', 'intern')}`);
-		assert.include(consoleStub.getCall(3).args[0], 'Should run in browser:');
-		assert.include(consoleStub.getCall(4).args[0], ' completed successfully');
+		assert.strictEqual(consoleStub.callCount, 4);
+		assert.include(consoleStub.getCall(0).args[ 0 ], 'testing "');
```
Comment (jason0x43) [inline tests/unit/runTests.ts]: `[ 0 ]` should presumably be `[0]` (same below)

File: tests/unit/runTests.ts [hunk 6]
```diff
@@ -122,108 +105,132 @@ describe('runTests', () => {
...
 				testingKey: 'key',
 				secret: 'secret'
 			});
-			assert.equal(args[2], 'tunnelOptions={ "verbose": "true", "apiKey": "key", "apiSecret": "secret" }');
-
-			assert.equal(
-				args[3],
-				'webdriver={ "host": "http://hub.testingbot.com/wd/hub", "username": "key", "accessKey": "secret" }'
-			);
+			assert.include(args, 'tunnelOptions={ "verbose": "true", "hostname": "hub.testingbot.com", "apiKey": "key", "apiSecret": "secret" }');
```
Comment (jason0x43) [inline tests/unit/runTests.ts]: Should be `username` and `accessKey`

File: tsconfig.json [hunk 7]
```diff
@@ -8,7 +8,11 @@
 		"sourceMap": true,
 		"strict": true,
 		"target": "es2016",
-		"types": [ "intern" ]
+		"types": [ "intern" ],
+		"typeRoots": [
```
Comment (jason0x43) [inline tsconfig.json]: Intern no longer has any custom types, so `typeRoots` isn't needed.

**Comment by rorticus** [comment]
@jason0x43 feedback addressed 👍
